bitcoin/src/parse.rs

```diff
@@ -84,6 +84,15 @@ pub(crate) fn int<T: Integer, S: AsRef<str> + Into<String>>(s: S) -> Result<T, P
     })
 }
 
+pub(crate) fn hex_u32<S: AsRef<str> + Into<String>>(s: S) -> Result<u32, ParseIntError> {
+    u32::from_str_radix(s.as_ref(), 16).map_err(|error| ParseIntError {
+        input: s.into(),
+        bits: u8::try_from(core::mem::size_of::<u32>() * 4).expect("max is 32 bits for u32"),
```

Comment:
This needs to be `* 8`: number of bits in a byte. (You could just put `32` there but it's nice to have it ready for making it generic.)

Reply:
Good catch